src/main/java/seedu/docit/logic/parser/prescription/DeletePrescriptionCommandParser.java

```diff
@@ -4,39 +4,41 @@
 
 import seedu.docit.commons.core.index.Index;
 import seedu.docit.logic.commands.prescription.DeletePrescriptionCommand;
+import seedu.docit.logic.parser.AppointmentParser;
 import seedu.docit.logic.parser.ArgumentMultimap;
 import seedu.docit.logic.parser.ArgumentTokenizer;
 import seedu.docit.logic.parser.CliSyntax;
-import seedu.docit.logic.parser.Parser;
 import seedu.docit.logic.parser.ParserUtil;
 import seedu.docit.logic.parser.exceptions.ParseException;
 
-public class DeletePrescriptionCommandParser implements Parser<DeletePrescriptionCommand> {
+public class DeletePrescriptionCommandParser implements AppointmentParser<DeletePrescriptionCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the DeletePrescriptionCommand and returns a
      * DeletePrescriptionCommand object for execution.
      *
      * @throws ParseException if the user input does not conform the expected format
      */
-    public DeletePrescriptionCommand parse(String args) throws ParseException {
+    public DeletePrescriptionCommand parseAppointmentCommand(String args) throws ParseException {
         Index index;
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_NAME);
 
-        try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeletePrescriptionCommand.MESSAGE_USAGE), pe);
-        }
-
         if (!ParserUtil.hasAllPrefixes(argMultimap, CliSyntax.PREFIX_NAME)
                 || argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeletePrescriptionCommand.MESSAGE_USAGE));
         }
 
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw pe;
+        }
+
         String medicineName = argMultimap.getValue(CliSyntax.PREFIX_NAME).get();
-        return new DeletePrescriptionCommand(index, medicineName);
 
+        if (medicineName.isBlank()) {
+            throw new ParseException("Medicine fields cannot be blank.");
```

Comment:
For code quality:
- Can make this a public static variable like: MESSAGE_FAILURE
- then can re-reference this same message for the DeletePrescriptionCommandParserTest